Updating to work with middleware

ElDuderino opened the pull request:
- Added dual serial output kludge (serDebug and serOutput)
- Added packet output for middleware
- Added mappings in SensorTypes.h (part of the Aretas API)
- Changed around the directory structure to include the necessary libraries to build (known working libs)

ElDuderino committed:
Updating to work with middleware

- Added dual serial output kludge (serDebug and serOutput)
- Added packet output for middleware
- Added mappings in SensorTypes.h (part of the Aretas API)

macleod-matt reviewed (approved):
Looks good. Does serialDebug->read read a single byte at a time like Serial.Read()? I dont see a definition. 

Your types defined in sensorTypes.h are greater than 1 byte. Could that pose a problem?

ElDuderino commented:
> Looks good. Does serialDebug->read read a single byte at a time like Serial.Read()? I dont see a definition.
> 
> Your types defined in sensorTypes.h are greater than 1 byte. Could that pose a problem?

Yeah I just replaced previous instances of Serial with a pointer to the Serial object so you can change it in a global scope by changing what the pointer points to (as long as it's an instance of HardwareSerial) 

They're defined by:
``HardwareSerial *serDebug = &Serial1;``
``HardwareSerial *serOutput = &Serial;``

Ideally, you could just re-define the pins for Serial1 to a pair of pins exposed on the expansion header of the CAN board like this (pins labeled from the adapter side):
```
 Serial1.end();
 Serial1.setRX(NEW_TX_PIN);
 Serial1.setTX(NEW_RX_PIN);
 Serial1.begin(115200);
```

Then you can hook up an FTDI adapter (or other UART->USB adapter) to view the UART output from Serial1 over USB. 

As for the type "byte" it shouldn't be a problem, but I do mention in the comments these are the first uint16 types we've tested in the API since we refactored so we should be on the lookout for any issues. The middleware will be fine given how we're emitting a comma delimited string and parse it into an int.
